packages/ra-ui-materialui/src/input/CheckboxGroupInput.spec.js

```diff
@@ -13,298 +13,191 @@ describe('<CheckboxGroupInput />', () => {
         );
-        const CheckboxElements = wrapper.find('WithStyles(FormControlLabel)');
-        const CheckboxElement1 = CheckboxElements.first();
-        assert.equal(CheckboxElement1.prop('label'), '**Male**');
+        expect(queryByLabelText('**John doe**')).not.toBeNull();
     });
 
     it('should not translate the choices if translateChoice is false', () => {
-        const wrapper = shallow(
+        const { queryByLabelText } = render(
             <CheckboxGroupInput
                 {...defaultProps}
-                choices={[
-                    { id: 'M', name: 'Male' },
-                    { id: 'F', name: 'Female' },
-                ]}
                 translate={x => `**${x}**`}
                 translateChoice={false}
             />
         );
-        const CheckboxElements = wrapper.find('WithStyles(FormControlLabel)');
-        const CheckboxElement1 = CheckboxElements.first();
-        assert.equal(CheckboxElement1.prop('label'), 'Male');
+        expect(queryByLabelText('**John doe**')).toBeNull();
+        expect(queryByLabelText('John doe')).not.toBeNull();
     });
 
     it('should displayed helperText if prop is present in meta', () => {
-        const wrapper = shallow(
+        const { queryByText } = render(
             <CheckboxGroupInput
                 {...defaultProps}
-                choices={[
-                    { id: 'M', name: 'Male' },
-                    { id: 'F', name: 'Female' },
-                ]}
-                translate={x => `**${x}**`}
-                translateChoice={false}
-                meta={{ helperText: 'Can i help you?' }}
+                meta={{ helperText: 'Can I help you?' }}
             />
         );
-        const FormHelperTextElement = wrapper.find(
-            'WithStyles(FormHelperText)'
-        );
-        assert.equal(FormHelperTextElement.length, 1);
-        assert.equal(
-            FormHelperTextElement.children().text(),
-            'Can i help you?'
-        );
+        expect(queryByText('Can I help you?')).not.toBeNull();
     });
 
     describe('error message', () => {
         it('should not be displayed if field is pristine', () => {
-            const wrapper = shallow(
+            const { container } = render(
                 <CheckboxGroupInput
                     {...defaultProps}
-                    choices={[
-                        { id: 'M', name: 'Male' },
-                        { id: 'F', name: 'Female' },
-                    ]}
-                    translate={x => `**${x}**`}
-                    translateChoice={false}
-                    meta={{ touched: false }}
+                    meta={{ touched: false, error: 'Required field.' }}
                 />
             );
-            const FormHelperTextElement = wrapper.find(
-                'WithStyles(FormHelperText)'
-            );
-            assert.equal(FormHelperTextElement.length, 0);
+            expect(container.querySelector('p')).toBeNull();
```

Comment:
You're testing implentation details here. If the root tag changes for some reason the test will fails but the user won't actually see anything. Those tests may become a pain to maintain. I believe testing the `Required field.` is not visible is enough

Reply:
You might use `FormHelperTextProps` to pass a `testid` prop if you really really want this

Reply:
This PR is already a great step in the right direction. I know I'm still testing implementation details every once in a while, but that's also because material-ui makes it somewhat hard. 

If the tests about helper text break in the future, I'll fix them. At least, tests about values and label won't break anymore. 

Many people are waiting for this change, so I suggest we merge it as is.